File: src/main/java/ch/scs/cc/testing/smells/Flight.java
package ch.scs.cc.testing.smells;

public class Flight {

    private final FlightNumber flightNumber;
    private boolean cancelled;

    public Flight(FlightNumber flightNumber) {
        this.flightNumber = flightNumber;
    }

    public FlightNumber getNumber() {
        return this.flightNumber;
    }

    public String getAirlineCode() {
        // TODO this is a stub
        return "";
    }

    public Object getAirline() {
        // TODO this is a stub
        return null;
    }

    public void setMileage(int value) {
        // TODO this is a stub
    }

    public int getMileageAsKm() throws InvalidRequestException {
        // TODO this is a stub
        if (cancelled) {
            throw new InvalidRequestException("Cannot get cancelled flight mileage");
        } else {
            return 1810;
        }
    }

    public void cancel() {
        // TODO this is a stub
        cancelled = true;
    }
}

File: CSharp/BackingCode/Flight.cs
namespace BackingCode
{

    using System;

    public class Flight
    {

        public string FlightNumber { get; }
        public string AirlineCode { get; }
        public object Airline { get; }
        public int Mileage { get; set; }

        public Flight(string flightNumber)
        {
            FlightNumber = flightNumber;
            AirlineCode = string.Empty;
        }

        public int MileageAsKm
        {
            // TODO this is a stub
            get { return 1810; }
        }
    }
}
